src/lib/components/data-graphic/marks/Line.svelte

```diff
@@ -0,0 +1,80 @@
+<!-- @component
+  Connects points together with a <path> element
+    in the order they appear in the data.
+-->
+<script lang="ts">
+  import { getContext, onDestroy } from "svelte";
+  import { extent } from "d3-array";
+
+  import {
+    lineFactory,
+    pathDoesNotDropToZero,
+  } from "$lib/components/data-graphic/utils";
+  import { guidGenerator } from "$lib/util/guid";
+  import { contexts } from "../constants";
+  import type { ExtremumResolutionStore, ScaleStore } from "../state/types";
+
+  const markID = guidGenerator();
+
+  export let data;
+  export let curve = "curveLinear";
+  export let xAccessor = "x";
+  export let yAccessor = "y";
+
+  export let color = "hsla(217,70%, 60%, 1)";
+  export let lineThickness = 1;
+  export let alpha = 1;
+  export let pathDefined = pathDoesNotDropToZero;
+
+  export let xMin = undefined;
+  export let xMax = undefined;
+  export let yMin = undefined;
+  export let yMax = undefined;
+
+  const xMinStore = getContext(contexts.min("x")) as ExtremumResolutionStore;
+  const xMaxStore = getContext(contexts.max("x")) as ExtremumResolutionStore;
+  const yMinStore = getContext(contexts.min("y")) as ExtremumResolutionStore;
+  const yMaxStore = getContext(contexts.max("y")) as ExtremumResolutionStore;
+
+  // get extents
+  $: [xMinValue, xMaxValue] = extent(data, (d) => d[xAccessor]);
+  $: [yMinValue, yMaxValue] = extent(data, (d) => d[yAccessor]);
+  // set your extrema here
+  $: xMinStore.setWithKey(markID, xMin || xMinValue);
```

Comment:
Would it make sense to have data exposed at the top level and not using the marks? Probably through a pluggable `Data` component or through `GraphicContext`? Plot components `Axis`, `Grid` etc. all rely on the same data. I am not sure if marks should be central authority over how data is being handled in all other components.

Having data more centrally would enable us to make multiple marks on a shared plot as well. So for something like an axis running from say 0-100 we can have one line from 0-40 and a completely different line running from 30-80 with shared axis, grid etc.

Reply:
I agree ~ I think we need to have it both ways, both as a top-level thing, and at the mark level. It's not always the case that the user  is operating with a single coherent data frame. The flexibility of being able to put data at either the shape or the plot level is pretty useful, for instance w/ `ggplot` you can put data at the top `ggplot` component and / or any of the marks, and the graphics in R will reconcile these plot bounds. Similar for us.